src/components/atoms/Tab/Tab.tsx

```diff
@@ -0,0 +1,31 @@
+import {Pressable} from 'react-native';
+import {NavIcon} from '../NavIcon';
+import {BottomTabNavigationEventMap} from '@react-navigation/bottom-tabs';
+import {ParamListBase, TabNavigationState} from '@react-navigation/routers';
+import {NavigationHelpers} from '@react-navigation/core';
+import {styles} from './styles';
+
+type TTabProps = {
+  state: TabNavigationState<ParamListBase>;
+  navigation: NavigationHelpers<ParamListBase, BottomTabNavigationEventMap>;
```

Comment:
możesz też useNavigation uzyć I guess, ja tam nie lubie tego przekazywania stanu przez propsy w rn-navigation, lmao, ale jak wolisz

Reply:
z tego co widziałem to niezbyt useNavigation zwraca coś innego i nie mam w nim dostępu np do ten funkcji emit co niżej używam

Reply:
masz dostęp, tylko musisz dobre typy podać, useNavigation ma różny typ w zależności od scope (czy jesteś w tabie, w drawerze, czy w stack'u np.)

Otypować to to zabawa, ale trza się tego nauczyć